cpp/src/arrow/io/file_test.cc

```diff
@@ -461,6 +461,11 @@ class MyMemoryPool : public MemoryPool {
     return Status::OK();
   }
 
+  Status ReallocateNoCopy(int64_t old_size, int64_t new_size, int64_t alignment,
+                          uint8_t** ptr) override {
+    return Reallocate(old_size, new_size, alignment, ptr);
```

Comment:
My comment above also implies that the implementation should be `return Status::NotImplemented()` so that caller immediately fallback to the `Reallocate()` with a size that is not necessarily the size that was passed to `TryResize`. The size passed to `TryResize` is conservative whereas the size passed to `Reallocate` is often more generous.